pkg/backend/apply.go

```diff
@@ -196,10 +197,13 @@ func PreviewThenPromptThenExecute(ctx context.Context, kind apitype.UpdateKind,
 	// Preview the operation to the user and ask them if they want to proceed.
 
 	if !op.Opts.SkipPreview {
-		changes, res := PreviewThenPrompt(ctx, kind, stack, op, apply)
+		plan, changes, res := PreviewThenPrompt(ctx, kind, stack, op, apply)
 		if res != nil || kind == apitype.PreviewUpdate {
 			return changes, res
 		}
+
+		// TODO(pdg-plan): should this check for an existing plan?
```

Comment:
We need more tests for how plans work when doing the auto preview as part of up. I'm not sure we should be replacing the plan here, but also because of mutation I don't think we can just run with the original plan object (*shakes fist at mutability by default*)